### ISSUE TITLE
Add dropout decay

### ISSUE THREAD

**Issue by firasdib** [deeplearning4j/deeplearning4j#3776]
#### Issue Description

It would be nice to be able to decay hyper parameters over time, for example starting with 40% dropout and slowly reducing it over the epochs to 10%.

**Comment by firasdib** [comment]
The reasoning behind this (besides trying to mimic the paper I'm reading) is that after reading [this](http://machinelearningmastery.com/dropout-regularization-deep-learning-models-keras/) article it would seem there is some sort of (linear?) dependency between learning rate and dropout.

He seems to argue that with a higher dropout you can consequently increase your learning rate. But if you as me have a decaying learning rate you'd want a decaying dropout rate to achieve the same thing.

Perhaps this could be implemented as a dropout-builder which you create once and attach to the layers (or the entire network).

**Comment by AlexDBlack** [comment]
Dropout now supports schedules - https://github.com/deeplearning4j/deeplearning4j/pull/3985
